lib/TemplatedPathPlugin.js

```diff
@@ -108,12 +123,14 @@ const replacePathVariables = (path, data) => {
 			.replace(REGEXP_ID, getReplacer(chunkId))
 			.replace(REGEXP_MODULEID, getReplacer(moduleId))
 			.replace(REGEXP_NAME, getReplacer(chunkName))
-			.replace(REGEXP_FILE, getReplacer(data.filename))
-			.replace(REGEXP_FILEBASE, getReplacer(data.basename))
+			.replace(REGEXP_FILE, getReplacer(escapePathVariables(data.filename)))
+			.replace(REGEXP_FILEBASE, getReplacer(escapePathVariables(data.basename)))
 			// query is optional, it's OK if it's in a path but there's nothing to replace it with
 			.replace(REGEXP_QUERY, getReplacer(data.query, true))
 			// only available in sourceMappingURLComment
 			.replace(REGEXP_URL, getReplacer(data.url))
+			.replace(/\[\\/g, "[")
+			.replace(/\\\]/g, "]")
```

Comment:
Replacing each on it's own can lead to false positives.

We also need to take care of the double escape case.

``` js
.replace(/\[\\(\\*[\w:]+\\*)\\\]/gi, "[$1]`)
```

Reply:
Good point, updated